waspc/e2e-test/GoldenTest.hs

```diff
@@ -83,3 +99,24 @@ runGoldenTest goldenTest = do
     writeFileManifest baseAbsFp filePaths manifestAbsFp = do
       let sortedRelativeFilePaths = unlines . sort . map (makeRelative baseAbsFp) $ filePaths
       writeFile manifestAbsFp sortedRelativeFilePaths
+
+    -- This function normalizes all package.json files for comparison as `npm install` can overwrite
+    -- them when creating/updating package-lock.json. Different versions of npm may produce different
+    -- (but semantically equivalent) package.json files, thus triggering false positives.
+    -- Ref: https://github.com/wasp-lang/wasp/issues/482
+    reformatPackageJsonFiles :: [FilePath] -> IO ()
+    reformatPackageJsonFiles allOutputFilePaths = do
+      let packageJsonFilePathSuffix = FP.pathSeparator : "package.json"
+      let packageJsonFilePaths = filter (packageJsonFilePathSuffix `isSuffixOf`) allOutputFilePaths
+      mapM_ reformatJson packageJsonFilePaths
+      where
+        aesonPrettyConfig = Config {confIndent = Spaces 4, confCompare = compare, confNumFormat = Generic, confTrailingNewline = True}
+        reformatJson jsonFilePath = do
+          let tmpFilePath = jsonFilePath ++ ".tmp"
+          str <- BSL.readFile jsonFilePath
+          -- NOTE: Aeson.decode into (:: Maybe Value) allows us to decode any
+          -- valid JSON string, without specifying a schema.
+          let json = fromJust (decode str :: Maybe Value)
+          let formattedJson = encodePretty' aesonPrettyConfig json
+          BSL.writeFile tmpFilePath formattedJson
+          renameFile tmpFilePath jsonFilePath
```

Comment:
Doing it with tmp is fine, but if you did a strict read up there, and not lazy, with readFile, then I think you wouldn't have to bother with tmp file, you could just write directly to the old file. Not 100% sure, but I think that should work.
You don't get any benefit with lazy reading here anyway.

Reply:
Yeah, I went down that path initially but hit a snag with `hGetContents` but can look quickly at it again. The tmp one has the nice property that the file write had to succeed first, and then renameFile which follows is atomic, but if a strict version of write failed part way I think we would corrupt our file, but that is probably not too likely. Will be good to just get strict working as a Haskell learning exercise anyway.

Reply:
Aha don't worry about it too much! I thought you can do it with https://hackage.haskell.org/package/bytestring-0.11.3.0/docs/Data-ByteString.html#g:29 though ?

Reply:
Ah, nice!! I'll give that a go :D I wasn't too keen on using something new but we already have that so that would be great 👍🏻 thnx